src/job/keygen.rs

```diff
@@ -0,0 +1,52 @@
+use std::{error::Error, fs::File, io::Write, path::Path};
+
+use rsa::{
+    pkcs8::{EncodePrivateKey, EncodePublicKey, LineEnding},
+    RsaPrivateKey, RsaPublicKey,
+};
+
+use crate::common::infra::config::CONFIG;
+
+/// Write to file if it doesn't exist.
```

Comment:
Nothing in this code that mentions the CA certificate. The code generates a public key certificate, but it's actually just a public key in PEM format. Typically, a certificate would be signed by a Certificate Authority (CA) or at least self-signed. This might not be an issue but could lead to misunderstandings about the nature of the generated files.